### PR TITLE
Add landmark to `<Hero>` component

### PR DESCRIPTION
**Pull request by TylerJDev**
## Summary

This will resolve the [accessibility engineering review for the `<Hero>` component](https://github.com/github/primer/issues/903). A labelled `<section>`element can help users navigate the page, as the label will describe the section's contents. 

## List of notable changes:

- Swaps out `<div>` for `<section>`, ties landmark with heading
- Adds prop `headingLevel` to change heading level (defaults to `h1`)

## Steps to test:

1. Inspect `<section>` region in DevTools
2. Ensure that `<section>` is tied to heading through `aria-labelledby`

## Supporting resources (related issues, external links, etc):

- [Region landmark example](https://www.w3.org/WAI/ARIA/apg/patterns/landmarks/examples/region.html)

## Contributor checklist:

- [ ] All new and existing CI checks pass
- [ ] Tests prove that the feature works and covers both happy and unhappy paths
- [ ] Any drop in coverage, breaking changes or regressions have been documented above
- [ ] New visual snapshots have been generated / updated for any UI changes
- [ ] All developer debugging and non-functional logging has been removed
- [ ] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

### CODE REVIEW

File: packages/react/src/Hero/Hero.tsx [hunk 1]
```diff
@@ -15,17 +15,26 @@ type Action = {
 
 export type HeroProps = BaseProps<HTMLDivElement> & {
   heading: string | React.ReactElement
+  headingLevel?: 'h1' | 'h2' | 'h3'
```
Comment (rezrah) [inline packages/react/src/Hero/Hero.tsx:18]: AFAIK the hero pattern should only ever use H1 and appear at the top of the page. We have other components like SectionIntro and CTABanner, which allow configurability for the other heading levels.

Is the configurability of the heading level super important right now? If not, wondering if we could defer this until the component rewrite has taken place. That way we can do this by simply forwarding the `as` in Heading.

File: packages/react/src/Hero/Hero.tsx [hunk 2]
```diff
@@ -24,8 +24,13 @@ export type HeroProps = BaseProps<HTMLDivElement> & {
 export const Hero = forwardRef<HTMLDivElement, HeroProps>(
   ({className, heading, description, primaryAction, secondaryAction, align = 'start', ...rest}, ref) => {
     return (
-      <div className={clsx(styles.Hero, styles[`Hero--align-${align}`], className)} ref={ref} {...rest}>
-        <Heading className={styles['Hero-heading']} as="h1">
+      <section
+        aria-labelledby="hero-section-brand-heading"
```
Comment (rezrah) [inline packages/react/src/Hero/Hero.tsx:28]: if someone wants to customise this, could we instead make this a prop value, with `hero-section-brand-heading` being the default value?

**Review by rezrah — APPROVED** [review]
Looks good, but also left some additional suggestions on improving customisation of the aria attribute value and making the change log entry more specific. Let me know if you want to talk through any of those points.

File: .changeset/gold-planes-call.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@primer/react-brand': patch
+---
+
+- Replaces `<div>` container with landmark element `<section>`
```
[the remark this answers is not in the document]
Comment (TylerJDev) [reply .changeset/gold-planes-call.md]: Love this! Thanks a bunch for the suggestion, will be more descriptive in the change logs!

File: packages/react/src/Hero/Hero.tsx (hunk 2 above)
Comment (TylerJDev) [reply packages/react/src/Hero/Hero.tsx]: That's a good call. I added a new prop as suggested, with the default.

**Pull request merged**